includes/ui/Notice.php

```diff
@@ -0,0 +1,226 @@
+	 *      color-info:      Blue highlight.
+	 *      color-strong:    Stronger border opacity highlight.
+	 *      type-hint:       Variation for a persistent hint.
+	 *      message-icon:    Show the icon next to the message instead of the title.
+	 *      alignwide:       Wide notice, with actions on the side.
+	 * }
+	 *
+	 * @return string HTML.
+	 */
+	public static function render( $options ) {
+
+		UI::ensure_styles();
+
+		$options = wp_parse_args(
+			$options,
+			[
+				'id'      => '',
+				'title'   => '',
+				'icon'    => '',
+				'message' => '',
+				'details' => '',
+				'buttons' => [],
+				'links'   => [],
+				'classes' => [],
+			]
+		);
+
+		$buttons = [];
+
+		foreach ( $options['buttons'] as $index => $button ) {
+			$primary   = $button['primary'] ?? ( 0 === $index );
+			$class     = $primary ? 'jm-notice__button' : 'jm-notice__button--outline';
+			$buttons[] = self::get_button_html( $button, $class );
+		}
+
+		$links = [];
+		foreach ( $options['links'] as $link ) {
+			$class   = $buttons ? 'jm-notice__button--link' : 'jm-notice__action';
+			$links[] = self::get_button_html( $link, $class );
+		}
+
+		$icon = self::get_icon_html( $options['icon'] );
+
+		ob_start();
+
+		$template_options = [
+			'options'      => $options,
+			'title'        => $options['title'],
+			'classes'      => $options['classes'] ?? [],
+			'message'      => $options['message'],
+			'details'      => $options['details'],
+			'icon_html'    => $icon,
+			'buttons_html' => $buttons,
+			'links_html'   => $links,
+		];
+
+		get_job_manager_template(
+			'notice.php',
+			$template_options
+		);
+
+		$notice_html = ob_get_clean();
+
+		/**
+		 * Filters notices. Return false to disable the notice.
+		 *
+		 * @since $$next-version$$
+		 *
+		 * @param string $notice_html Generated HTML for the notice.
+		 * @param array  $options Notice template options.
+		 */
+		$notice_html = apply_filters( 'wpjm_notice', $notice_html, $template_options );
+
+		if ( ! empty( $options['id'] ) ) {
+			/**
+			 * Filters an individual notice.Return false to disable the notice.
+			 *
+			 * @since $$next-version$$
+			 *
+			 * @param string $notice_html Generated HTML for the notice.
+			 * @param array  $options Notice template options.
+			 */
+			$notice_html = apply_filters( 'wpjm_notice_' . $options['id'], $notice_html, $template_options );
+		}
+
+		return is_string( $notice_html ) ? $notice_html : '';
+	}
+
+	/**
+	 * Generate HTML for a notice icon.
+	 *
+	 * @param string|null $icon Icon name or a safe SVG code.
+	 *
+	 * @return string Icon HTML.
+	 */
+	private static function get_icon_html( $icon ) {
+		$html = '';
+
+		if ( ! empty( $icon ) ) {
+
+			$is_classname = preg_match( '/^[\w-]+$/i', $icon );
+
+			$html = '<div class="jm-notice__icon"';
+
+			if ( $is_classname ) {
+				$html .= ' data-icon="' . esc_attr( $icon ) . '"';
+			}
+
+			$html .= '>';
+
+			if ( ! $is_classname ) {
+				$html .= ( $icon );
```

Comment:
Does it need to be escaped ?

Reply:
This is for using SVG code for the icon, which makes it a bit tricky in WP. Added a wp_kses call now covering the basics.